libfwupdplugin/fu-archive.c

```diff
@@ -162,7 +185,7 @@ fu_archive_load(FuArchive *self, GBytes *blob, FuArchiveFlags flags, GError **er
 		return FALSE;
 	}
 	while (TRUE) {
-		const gchar *fn;
+		const gchar *fn = NULL;
```

Comment:
Will revert. the change was for the function `_save()`. Now I understand why it looped forever before I force `*filename = NULL` in the callback :facepalm: